run_md.py

```diff
@@ -0,0 +1,879 @@
+
+
+def continue_md(tpr, cpt, xtc, wdir, new_mdtime_ps, deffnm_next, project_dir):
+    try:
+        subprocess.check_output(f'wdir={wdir} tpr={tpr} cpt={cpt} xtc={xtc} new_mdtime_ps={new_mdtime_ps} '
+                                f'deffnm_next={deffnm_next} bash {os.path.join(project_dir, "continue_md.sh")}', shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(f'{wdir}\t{e}\n')
+        return None
+
+    return wdir
+
+
+
+def main(protein, wdir, lfile=None, system_lfile=None,
+         forcefield_num=6, addH=True, clean_previous=False,
+         gromacs_version="GROMACS/2021.4-foss-2020b-PLUMED-2.7.3",
+         mdtime_ns=1, npt_time_ps=100, nvt_time_ps=100,
+         topol=None, topol_itp_list=None, posre_list_protein=None,
+         wdir_to_continue_list=None,
+         tpr_prev = None, cpt_prev=None,
+         xtc_prev=None,
+         deffnm_prev='md_out',
+         hostfile=None, ncpu=1,
+         not_clean_log_files=False):
+    '''
+
+    :param protein: protein file - pdb or gro format
+    :param wdir: None or path
+    :param lfile: None or file
+    :param system_lfile: None or file. Mol or sdf format
+    :param forcefield_num: int
+    :param addH: boolean. Add hydrogens by default
+    :param clean_previous: boolean. Remove all previous md files
+    :param gromacs_version:
+    :param mdtime_ns: float. Time in ns
+    :param npt_time_ps: int. Time in ps
+    :param nvt_time_ps: int. Time in ps
+    :param topol: None or file
+    :param topol_itp_list: None or list of files
+    :param posre_list_protein: None or list of files
+    :param wdir_to_continue_list: list of paths
+    :param tpr_prev: None or file
+    :param cpt_prev: None or file
+    :param xtc_prev: None or file
+    :param deffnm_prev: md_out
+    :param hostfile: None or file
+    :param ncpu:
+    not_clean_log_files: boolean. Remove backup md files (starts with #)
+    :return:
+    '''
+
+
+    try:
+        subprocess.check_output(f'module load {gromacs_version}', shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(e)
+        return None
+
+    project_dir = os.path.dirname(os.path.abspath(__file__))
+    script_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'scripts')
+
+    if wdir_to_continue_list is None:
+        # create dirs
+        wdir_protein = os.path.join(wdir, 'md_preparation', 'protein')
+        wdir_ligand = os.path.join(wdir, 'md_preparation', 'var_lig')
+        wdir_system_ligand = os.path.join(wdir, 'md_preparation', 'system_lig')
+
+        wdir_md = os.path.join(wdir, 'md_preparation', 'md_files')
+
+        os.makedirs(wdir_md, exist_ok=True)
+        os.makedirs(wdir_protein, exist_ok=True)
+        os.makedirs(wdir_ligand, exist_ok=True)
+        os.makedirs(wdir_system_ligand, exist_ok=True)
+
+        if not os.path.isfile(protein):
+            raise FileExistsError(f'{protein} does not exist')
```

Comment:
You may check this inside `filepath_type`. The same function also may check for files extensions.

for example:
```python
def filepath_type(x, ext=None, check_exist=True):
    value = os.path.abspath(x) if x else x
    if check_exist and not os.path.isfile(value):
        raise FileExistsError(f'{value} does not exist')
    if ext and os.path.splitext(value)[1] not in ext:
        raise FileExistsError(f'File {value} has a wronh extension. Allowed are {", ".join(ext)}')
    return value
```

and call in `argparse` them using `partial`
```python
type=partial(filepath_type, ext=('mol', 'sdf', 'gro'), check_exist=True)
```

Reply:
or you may use parser.error approach to report about issues
https://stackoverflow.com/questions/11540854/file-as-command-line-argument-for-argparse-error-message-if-argument-is-not-va